src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEPropertySymbol.cs

```diff
@@ -172,7 +172,12 @@ private enum Flags : byte
 
             // Decode nullable before tuple types to avoid converting between
             // NamedTypeSymbol and TupleTypeSymbol unnecessarily.
-            propertyTypeWithAnnotations = NullableTypeDecoder.TransformType(propertyTypeWithAnnotations, handle, moduleSymbol);
+
+            // The containing type is passed to NullableTypeDecoder.TransformType to determine access
+            // (when synthesizing nullable state for a private member in a [NullableContext] but without
+            // explicit [Nullable]) because the accessibility of the property has not been calculated yet.
+            // At worst, this means we have incorrect nullability info for inaccessible properties.
```

Comment:
>// At worst, this means we have incorrect nullability info for inaccessible properties. [](start = 12, length = 87)

Given complexity of calculation of accessibility for properties (overriding factored in, etc.), from the usability perspective (what other tools would have to do to calculate it), it might be better to always delegate to type's accessibility even on emit. Basically only accessibility of entities that have their accessibility explicitly encoded in metadata should be taken into account for the purpose of filtering. Events probably fall into the same category.